### PR TITLE
Refactor main loops. 

### LINKED ISSUE #81: [new_charts] refactor outer loop
As mentioned previously, if you pass a non-existing key to the plotting script's `-b`, then no graphs are processed.

Before we fix this, I recommend refactoring the outer loops, otherwise we have to apply a hack twice.

### PR DESCRIPTION
**Pull request by snim2**
Replace assertions with custom error function. Add error message if argument to -b does not appear in Krun results files. [plot_krun_results]

Fixes #81

### CODE REVIEW

File: plot_krun_results.py [hunk 1]
```diff
@@ -85,74 +86,64 @@ def main(is_interactive, data_dcts, window_size, outfile, xlimits, single_exec,
...
-                                                     mean,
-                                                     sigma)
-                        fig.set_size_inches(*export_size)
-                        pdf.savefig(fig,
-                                    dpi=fig.dpi, orientation='landscape',
-                                    bbox_inches='tight')
-                        plt.close()
-        print('Saved: %s' % outfile)
+                else:
+                    fig, export_size = draw_page(is_interactive,
```
Comment (vext01) [inline plot_krun_results.py]: Do we need the call to drawpage() twice? Perhaps we can always assign the return values, but only use them in one mode?

**Comment by vext01** [comment]
Looks good apart from one minor nit I commented upon.

**Comment by vext01** [comment]
(tested on bencher2 headless+PDF and with interactive/X-forwarding)

**Comment by snim2** [comment]
OK, just testing a fix for that comment.

**Comment by snim2** [comment]
Ready to squash?

**Comment by vext01** [comment]
Looks good, go ahead.

**Comment by snim2** [comment]
Splat.

**Comment by vext01** [comment]
Many thanks!

**Pull request merged**